### DISCUSSION TITLE
Seed step change?

### DISCUSSION THREAD

**Discussion by DavidKouw** [lllyasviel/Fooocus#2597] [Q&A] [answered]
Is there any way to change the seed increase parameter? It is currently set to 1. But what if I want the seed to change by a specified number every time?

**Comment by mashb1t** [comment]
There is no such functionality in the UI, seeds are always increased by one if not disabled in developer debug mode. You can always manually generate with increment 2 or adjust the code accordingly. Why would you want to have / use this feature?

**Reply by DavidKouw** [reply]
@mashb1t  No no no, this feature doesn't need to be in the UI. It is not something that may be needed on a regular basis. 

What's the point: Pony-based models are quite unruly, and often only give me what I need after a long series of generations. In the process, I discovered that there is a behavioral pattern in which the model generates 2-5 images in a row that are not very different from each other. Then the influence of tokens somehow changes and a number of similar generations follow, but with the predominance of some other specific token. Thus it turns out that one query can get several conceptually different results. But for each such result there will be several slightly different images (as if I used Vary (Subtle)).  
That's why when it's time to experiment I'd like to set the step for a seed to be greater than 1, in order to skip similar generations. 

I would be grateful if you could just tell me in which file I can change this parameter myself =)

**Reply by mashb1t** [reply, marked as answer]
You can change the increment here: https://github.com/lllyasviel/Fooocus/blob/main/modules/async_worker.py#L440

**Reply by DavidKouw** [reply]
Thanks! Insanely grateful for the guide and in general for the work on Fooocus. You are the best!

**Comment by poor7** [comment]
For a random seed, I use the endless generation mode, activated by right-clicking on the Generate button.

**Reply by DavidKouw** [reply]
Yeah, it works BUT -> I don't have the most powerful hardware and I don't have the energy or desire to spend time on long series of slightly different images. In that case, a larger seed pitch is a lifesaver for me.

**Comment by ustas-eth** [comment]
@mashb1t I had the same question and it would actually be good to put the parameter under the developer settings in the UI

**Reply by mashb1t** [reply]
referenced in https://github.com/lllyasviel/Fooocus/issues/2777

**Reply by DavidDragonsage** [reply]
Wonderful! Thank you. 🙂

**Comment by DavidDragonsage** [comment]
Although I would not consider it a priority, I would also find it useful to have a different seed step value available in developer settings. I am in a similar boat to DavidKouw in that my hardware is poor and my patience to look at 100+ images after an overnight run is limited.

Getting Fooocus running and setup always takes significant time in itself (it only works every second time I start Fooocus, guaranteed), so the last thing I would want to do is to routinely modify code and then go through the process again.
